tests/src/Kernel/Entity/GetEntityGroupsTest.php

```diff
@@ -0,0 +1,216 @@
+    $this->assertTrue(Og::isMember($this->group1, $this->user1));
+    $this->assertFalse(Og::isMember($this->group1, $this->user2));
+
+    $actual = Og::getEntityGroups($this->user2);
+
+    $this->assertCount(1, $actual['entity_test']);
+    $this->assertGroupExistsInResults($this->group2, $actual);
+
+    // Also check isMember.
+    $this->assertTrue(Og::isMember($this->group2, $this->user2));
+    $this->assertFalse(Og::isMember($this->group2, $this->user1));
+  }
+
+  /**
+   * Tests other groups users are added to.
+   */
+  public function testOtherGroups() {
+    // Should be a part of no groups.
+    $this->assertEquals([], Og::getEntityGroups($this->user3));
+    $this->assertFalse(Og::isMember($this->group1, $this->user3));
+    $this->assertFalse(Og::isMember($this->group2, $this->user3));
+
+    // Invalidate the caches so the static cache is cleared and group data is
+    // fetched again for the user.
+    Og::invalidateCache();
+
+    // Add user to group 1 should now return that group only.
+    $this->createMembership($this->user3, $this->group1);
+
+    $actual = Og::getEntityGroups($this->user3);
+
+    $this->assertCount(1, $actual['entity_test']);
+    $this->assertGroupExistsInResults($this->group1, $actual);
+
+    $this->assertTrue(Og::isMember($this->group1, $this->user3));
+    $this->assertFalse(Og::isMember($this->group2, $this->user3));
+
+    Og::invalidateCache();
+
+    // Add to group 2 should also return that.
+    $this->createMembership($this->user3, $this->group2);
+
+    $actual = Og::getEntityGroups($this->user3);
+
+    $this->assertCount(2, $actual['entity_test']);
+    $this->assertGroupExistsInResults($this->group1, $actual);
+    $this->assertGroupExistsInResults($this->group2, $actual);
+
+    $this->assertTrue(Og::isMember($this->group1, $this->user3));
+    $this->assertTrue(Og::isMember($this->group2, $this->user3));
+  }
+
+  /**
+   * Creates an Og membership entity.
+   *
+   * @todo This is a temp function, which will be later replaced by Og::group().
+   *
+   * @param \Drupal\user\Entity\User $user
+   * @param \Drupal\Core\Entity\EntityInterface $group
+   *
+   * @return \Drupal\og\Entity|OgMembership
+   */
+  protected function createMembership($user, $group) {
+    $membership = OgMembership::create(['type' => OG_MEMBERSHIP_TYPE_DEFAULT])
+      ->setEntityId($user->id())
+      ->setEntityType('user')
+      ->setGid($group->id())
+      ->setGroupType($group->getEntityTypeId())
+      ->save();
+
+    return $membership;
+  }
+
+  /**
+   * Asserts whether a group ID exists in some results.
+   *
+   * Assumes entity_type is used.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $group_to_check
+   * @param array $results
+   */
+  protected function assertGroupExistsInResults($group_to_check, array $results) {
+    $found = FALSE;
+    foreach ($results['entity_test'] as $group) {
+      if ($group->id() == $group_to_check->id()) {
+        $found = TRUE;
```

Comment:
Instead of this assertTrue() IMHO it would be better to throw an exception if we reach the end of loop without an early return